File: src/main/java/com/dev/farmmanager/service/cropcycle/CropCycleService.java
package com.dev.farmmanager.service.cropcycle;

import com.dev.farmmanager.domain.dto.cropcycle.CropCyclePageDto;
import com.dev.farmmanager.domain.entity.CropCycle;
import com.dev.farmmanager.domain.enumeration.CropCycleStatus;
import com.dev.farmmanager.domain.payload.cropcycle.CropCyclePayload;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface CropCycleService {

    CropCyclePageDto findAll(CropCycleStatus status, LocalDate date);

    Optional<CropCycle> getById(Integer id);

    CropCycle create(Integer userId, CropCyclePayload payload);

    CropCycle update(Integer id, CropCyclePayload payload);

    void delete(Integer id);
}

File: frontend/src/services/crop-cycle.service.ts
import http from '@/api/http'
import type { CropCycleDto, CropCyclePage, CropCyclePayload, CropCycleStatus } from '@/types/crop-cycle'

export interface CropCycleFilters {
  status?: CropCycleStatus | null
  date?: string | null
}

export const cropCycleService = {
  findAll: (filters?: CropCycleFilters) => {
    const params: Record<string, string> = {}
    if (filters?.status) params.status = filters.status
    if (filters?.date)   params.date   = filters.date
    return http.get<CropCyclePage>('/farm/crop-cycle', { params }).then((r) => r.data)
  },

  create: (payload: CropCyclePayload) =>
    http.post<CropCycleDto>('/farm/crop-cycle', payload).then((r) => r.data),

  getById: (id: number) =>
    http.get<CropCycleDto>(`/farm/crop-cycle/${id}`).then((r) => r.data),

  update: (id: number, payload: CropCyclePayload) =>
    http.put<CropCycleDto>(`/farm/crop-cycle/${id}`, payload).then((r) => r.data),

  delete: (id: number) =>
    http.delete(`/farm/crop-cycle/${id}`),
}
